src/service/position-observer/position-observer-impl.js

```diff
@@ -84,6 +85,13 @@ export class PositionObserver {
     }
 
     worker.update();
+
+    return () => {
```

Comment:
let's return the existing `unobserve` (which also stops all monitoring when last worker is gone)

Reply:
why can't we just use `unobserve(element)`?  Is it because now we need to call `#observe()` on single element multiple times? Otherwise I don't see a requirement to return a unlistener

Reply:
@zhouyx Yep, the position of an element may be observed more than once. `unobserve` is an unsafe operation, as it will stop the *first* registered worker for an element.

Updating the unlistener to stop monitoring is trivial.

Reply:
i see, could we add a new `removeWorker_(worker) ` and have this call it and also have `unobserve` call it for all workers that match the element as well? This way the logic that stops listening when no workers is left can be shared.

Reply:
My other question is why do observe an element more than once? instead of having a list of handlers passed when we call `#observe()`

Personally I think it is very confusing that `unobserve()` and `unlisten()` does different things.

Reply:
Different modules may choose to observe the element independently. I'm all for having a single worker for an element, but `PositionObserver` doesn't have the proper tools to handle that.

An example where two position observers may be need for the same element is video, where `autoplay` requires an observable for visibility and `docking` requires an observable for actual position. These features are decoupled from each other so it makes sense that they observe the same element independently.

Reply:
I think it is worth removing `unobserve` and having all places that use it call the function that is returned from `observe`. Maybe an issue separate PR for it?

Reply:
That makes sense. Let's remove `unobserve()`